Library/Homebrew/cmd/search.rb

```diff
@@ -64,7 +67,9 @@ def search
     end
 
     if args.remaining.empty?
-      if args.casks?
+      if args.formulae?
```

Comment:
Might want to think here what it should do if `--formulae` and `--casks` is passed? I'd suggest a `raise UsageError, "Cannot specify --formulae and --casks without <text>!`

Reply:
On it.